src/js/administration/actions.js

```diff
@@ -30,7 +30,6 @@ export const updateSetting = (key, value) => {
  * @param update {object} update data of key-value pairs
  * @returns {object}
  */
-export const updateSettings = update => ({
-    type: UPDATE_SETTINGS.REQUESTED,
-    update
-});
+export const updateSettings = createAction(UPDATE_SETTINGS.REQUESTED, update => ({
+    payload: { update }
+}));
```

Comment:
Agreed. There are a few occasions that the createAction is expecting to get a single object where we could avoid nested objects. Not sure for the specific case of update, but for some of the cases it can get complicated by the fact that the object can be passed by itself or with another variable depending on what the action type is. Not the end of the world but it could add some confusion in the reducer if update can be referred to by `action.payload` or `action.payload.update` depending on the action type. Definitely still worth looking into though.

Reply:
I've created low priority issue for this.